### PR TITLE
Remove deprecated RenderEditable.onSelectionChanged

### PR DESCRIPTION
**Pull request by Piinks**
This removes `onSelectionChanged` from `RenderEditable`

This deprecated property has reached end of life after the release of Flutter 2.10

- The replacement is `textSelectionDelegate.userUpdateTextEditingValue`.
- ❌  This migration is not supported by the dart fix tool. 
- 🕐  This API was deprecated in https://github.com/flutter/flutter/pull/76653, first tagged v1.26

Part of https://github.com/flutter/flutter/issues/98537
For the full list of deprecations being removed in this batch, see [flutter.dev/go/deprecations-removed-after-2-10](https://docs.google.com/spreadsheets/d/12krawYCu6X_g_5wLGpmiAIi-VcTRV6xG7RGbRovuatQ/edit?usp=sharing)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by Piinks** [comment]
There are some framework tests here that need to be updated

**Comment by Piinks** [comment]
Also remove the deprecated TextSelectionDelegate setter textEditingValue in this PR, they are related.

File: packages/flutter/test/rendering/editable_test.dart [hunk 1]
```diff
@@ -484,6 +484,7 @@ void main() {
       // This makes the scroll axis vertical.
       maxLines: 2,
       textSelectionDelegate: delegate,
+      // TODO(Piinks): Migrate these tests
```
Comment (Piinks) [inline packages/flutter/test/rendering/editable_test.dart:487]: @chunhtai is there a migration guide for this? Can you advise how to migrate?

Comment (chunhtai) [reply packages/flutter/test/rendering/editable_test.dart:487]: you can create override userUpdateTextEditingValue in _FakeEditableTextState to listen to the selection changes

File: packages/flutter/lib/src/rendering/editable.dart [hunk 2]
```diff
@@ -1981,6 +1940,10 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin,
       extentOffset: extentOffset,
       affinity: fromPosition.affinity,
     );
+    final bool focusingEmpty = newSelection.baseOffset == 0 && newSelection.extentOffset == 0 && !hasFocus;
+    if (newSelection == selection && cause != SelectionChangedCause.keyboard && !focusingEmpty) {
```
Comment (chunhtai) [inline packages/flutter/lib/src/rendering/editable.dart:1944]: why does it need to check for other condition if the selection is the same?

Comment (Piinks) [reply packages/flutter/lib/src/rendering/editable.dart:1944]: I am not sure, I found this change that introduced this: https://github.com/flutter/flutter/pull/29563
cc @mdebbar 
It was only calling onSelectionChanged under those circumstances.

Without this change, this test from that change fails:
https://github.com/flutter/flutter/blob/9cfaabaed09e1c6606c9148b0f716a78146ec4e6/packages/flutter/test/rendering/editable_test.dart#L725


However, I notice it now causes another failure, so there may be a bug here. One test was expecting one thing from the old API, and the other test is expecting something else from the new API.

Comment (Piinks) [reply packages/flutter/lib/src/rendering/editable.dart:1944]: I assumed the logic in `_handleSelectionChange` was necessary, so I carried it over to the new API. It probably was modified by other changes after #29563. Let me see if removing the keyboard and focus part still passes the existing test.

Comment (mdebbar) [reply packages/flutter/lib/src/rendering/editable.dart:1944]: The `!focusingEmpty` was added in https://github.com/flutter/flutter/pull/49000.

The `cause != SelectionChangedCause.keyboard` was added in https://github.com/flutter/flutter/pull/42879.

Comment (Piinks) [reply packages/flutter/lib/src/rendering/editable.dart:1944]: 🙏  Thanks for tracing those back @mdebbar! So I think this is indeed intentional behavior, I am looking at the other test now that is broken here by carrying the logic over.

Comment (Piinks) [reply packages/flutter/lib/src/rendering/editable.dart:1944]: https://github.com/flutter/flutter/pull/29809/files is where the test came from, the cupertino one is broken, while the same material test passes, so we may be missing something in cupertino text field.

**Comment by Piinks** [comment]
Touched base with @justinmc, it looks like the flickering bug #29563 fixed does not exist with the new API. As I understand it, the repeated calls to onSelectionChanged caused the flickering, and so removing `onSelectionChanged` makes sense to no longer reproduce the bug. I have tested this on Android and iOS and do not see any flickering, so I am going to remove the test.
Thanks all for taking a look. Let me know if there are additional repros or any concerns about a regression here.

**Review by mdebbar — APPROVED** [review]
LGTM

**Review by chunhtai — APPROVED** [review]
LGTM

**Pull request merged**